File: lib/ts/src/internal/types/TFieldDeclaration.ts
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

import { TTypeDeclaration } from './TTypeDeclaration.js';

export class TFieldDeclaration {
    fieldName: string;
    typeDeclaration: TTypeDeclaration;
    optional: boolean;

    constructor(fieldName: string, typeDeclaration: TTypeDeclaration, optional: boolean) {
        this.fieldName = fieldName;
        this.typeDeclaration = typeDeclaration;
        this.optional = optional;
    }
}

File: lib/go/internal/types/TFieldDeclaration.go
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package types

// TFieldDeclaration describes a struct field in a Telepact schema.
type TFieldDeclaration struct {
	FieldName       string
	TypeDeclaration *TTypeDeclaration
	Optional        bool
}

// NewTFieldDeclaration constructs a TFieldDeclaration instance.
func NewTFieldDeclaration(fieldName string, typeDeclaration *TTypeDeclaration, optional bool) *TFieldDeclaration {
	return &TFieldDeclaration{
		FieldName:       fieldName,
		TypeDeclaration: typeDeclaration,
		Optional:        optional,
	}
}
